src/pint/plot_utils.py

```diff
@@ -176,3 +177,127 @@ def phaseogram_binned(
         plt.close()
     else:
         plt.show()
+
+
+def plot_priors(
+    model,
+    chains,
+    fitvals,
+    fiterrs,
+    maxpost_fitvals,
+    burnin=100,
+    bins=100,
+    scale=False,
+    file=False,
+):
+
+    """
+    Show binned samples, prior probability distribution and an initial gaussian
+    probability distribution plotted with 2 sigma, maximum posterior and original
+    fit values marked.
+    
+    Parameters
+    ----------
+    model
+        Input timing model 
+    chains: dict    
+        Post MCMC integration chains that are input into a histogram and compared with the priors. Contain the fitkeys and the samples
+    fitvals
+        Initial Parameter values
+    fiterrs
+        Initial Parameter uncertainties
+    maxpost_fitvals
+        Maximum Posterior values
+    With scale is True
+        The priors and initial probability distributions are scaled to the peak of the histogram.
+    With scale is False
+        The priors and initial probability distributions are plotted without the scaling.
+    """
+    keys = []
+    values = []
+    for item in chains.items():
+        keys.append(item[0]), values.append(item[1])
+
+    priors = []
+    initials = []
+    x_range = []
+    counts = []
+    for i in range(0, len(keys[:-1])):
+        values[i] = values[i][burnin:].flatten()
+        x_range.append(np.linspace(values[i].min(), values[i].max(), num=bins))
+        priors.append(getattr(model, keys[i]).prior.pdf(x_range[i]))
+        initials.append(
+            GaussianBoundedRV(loc=float(fitvals[i]), scale=float(fiterrs[i])).pdf(
+                x_range[i]
+            )
+        )
+        a, x = np.histogram(values[i], bins=bins)
+        counts.append(a)
+
+    fig, axs = plt.subplots(len(keys), figsize=(8, 11), constrained_layout=True)
+
+    for i, p in enumerate(keys):
+        if i != len(keys[:-1]):
```

Comment:
Should this be `len(keys)-1`? This might behave a little different with a zero-length list of keys, but otherwise the subtraction is probably clearer. Or, actually, is this really supposed to run for all but the last key?

Reply:
It is supposed to run for all but the last key.